### DISCUSSION TITLE
Hidden splashscreen

### DISCUSSION THREAD

**Discussion by donlaiq** [AppFlowy-IO/AppFlowy#3000] [Ideas]
Hi,

I was playing around with the source code, and when I run the application, I saw something weird for a fraction of a second. It was hard to guess what it was, so I run it again, and I saw it again.
Finally, after digging into the code, I saw that the app uses a splash screen, but I wasn't able to contemplate it because of the speed it was disappearing.

I don't want to open a pull request with it, because finally it's a matter of taste, and the functionality to change it is already there, but it would be nice to see the splash screen for a few seconds.

I'm attaching a video where I've made some changes to have a constant duration, and now you have some time to see it. What do you think about it? Do you have this issues on another OS?

https://github.com/AppFlowy-IO/AppFlowy/assets/10087643/7a5fd5a9-e074-40c3-9882-0d6af82cb649

**Comment by Xazin** [comment]
We use Navigator.push instead of Navigator.pushReplacement.

**Reply by donlaiq** [reply]
I don't understand your answer @Xazin. Maybe, you are trying to read the code from the video, but please ignore it. I just recorded the rectangle where the window opens, but the code behind is just a funny coincidence.
I'm just saying that with the current version of AppFlowy,  I am not able to see the splash screen (next video), but after doing a few changes, and play with some variables/constants, I got the version from the previous video. As I said before, it's just a matter of taste.

https://github.com/AppFlowy-IO/AppFlowy/assets/10087643/d13e24c4-f332-4bc7-97b3-83d457953094

**Reply by Xazin** [reply]
> I don't understand your answer @Xazin. Maybe, you are trying to read the code from the video, but please ignore it. I just recorded the rectangle where the window opens, but the code behind is just a funny coincidence.
> 
> I'm just saying that with the current version of AppFlowy,  I am not able to see the splash screen (next video), but after doing a few changes, and play with some variables/constants, I got the version from the previous video. As I said before, it's just a matter of taste.
> 
> 
> 
> https://github.com/AppFlowy-IO/AppFlowy/assets/10087643/d13e24c4-f332-4bc7-97b3-83d457953094
> 
> 
> 
> 

Sorry, my reply was for other developers.

When we show the splash screen, we push a new screen on top of it, instead of replacing the current screen. This means that the first item in the navigation stack is always the splash screen.

**Reply by donlaiq** [reply]
Sure it is... but it disappears at light speed :)
